version: '3.4'

services:
  athelo-backend-api:
    build: ./rest-api
    restart: always
    image: athelo-backend
    container_name: athelo-backend-api
    volumes: 
      - ./rest-api:/app
      - ./application_default_credentials.json:/application_default_credentials.json
    ports:
      - 8080:8080
    environment:
    - ENVIRONMENT=local
    - GOOGLE_APPLICATION_CREDENTIALS=/application_default_credentials.json
    - GOOGLE_CLOUD_PROJECT=test-deploy-402816
    stdin_open: true
    tty: true 
    # command: tail -f /dev/null
    command: python3 app.py
    depends_on:
      - redis
    networks:
      - athelo_network
  ws-server:
    build: ./ws-server
    container_name: ws-server
    restart: on-failure
    depends_on:
      - athelo-backend-api
      - redis
    volumes:
      - ./ws-server:/flask_app
    networks:
      - athelo_network
  db:
    image: postgres:15
    container_name: postgres
    volumes:
      - postgres_data:/var/lib/postgresql/data/
    environment:
      - POSTGRES_USER=athelo
      - POSTGRES_PASSWORD=athelo
      - POSTGRES_DB=athelo
    networks:
      - athelo_network
  redis:
    image: sameersbn/redis:4.0.9-3
    container_name: redis
    ports:
      - "6379:6379"
    environment:
      REDIS_PASSWORD: redispassword
    volumes:
      - redis_data:/var/lib/redis
    restart: always
    networks:
      - athelo_network
  nginx:
    image: nginx:1.25.3-alpine
    container_name: nginx
    restart: on-failure
    volumes:
      - ./nginx/nginx.conf:/etc/nginx/nginx.conf
    ports:
      - 8000:80
    depends_on:
      - athelo-backend-api
      - ws-server
    networks:
      - athelo_network

volumes:
  postgres_data:
  redis_data:

networks:
  athelo_network:
    driver: bridge
